app.js

```diff
@@ -10,6 +31,141 @@ function Book(title, author, pages, read) {
+  });
+}
+
+// function for handling click event on Submit.
+submitBook.addEventListener("click", selectOperation);
+
+  function render(arr){
+    let html = "";
+    for (let book of arr) {
+      let readingStatus = book.read == "read" ? "You have read this book" : "You haven't read this book";
+      html += `<div class= "book-card"><h3 class="book-title">${book.title}</h3>
+              <p class="author-name">by ${book.author}</p>
+              <p class="description">${book.description}</p>
+              <div class="book-details">
+                <p class="book-pages"><i class="fa fa-chevron-right"></i> No. of Pages: <span>${book.pages}</span></p>
+                  <p><i class="fa fa-chevron-right"></i> ${readingStatus}</p>
+              </div>
+              <div class="btns">
+                <button class="btn-read-status book-buttons" onclick="changeStatus('${book.id}')">Read</button>
+                <button class="btn-remove book-buttons" onclick="deleteBook('${book.id}')">Delete</button>
+                <button class="btn-edit book-buttons" onclick="editBookInfo('${book.id}')">Edit</button>
+              </div>
+          </div>`;
+    }
+    // document.querySelector('#book-list').insertAdjacentHTML('beforeend', html);
+    document.getElementById('book-list').innerHTML = html;
+}
+
+// Validates the input form.
+function inputValid(a, b, c, d) {
+  if (a == "" || b == "" || c == "" || d == "") {
+    alert('You need to fill all the form');
+    return false;
+  }
+  return true;
+}
+// Generate random IDs for books.
+function randomId() {
+  return '_' + Math.random().toString(36).substr(2, 9);
+}
+
+//Delete book from the array.
+function deleteBook(id) {
+  let index = getIndexofBook(id);
+  myLibrary.splice(index, 1);
+  localStorage.setItem("myLibrary", JSON.stringify(myLibrary));
+  render(myLibrary);
+}
+
+// Changes the reading status.
+function changeStatus(id){
+  let index = getIndexofBook(id);
+  myLibrary[index].read = myLibrary[index].read == 'read' ? 'unread' : 'read';
+  localStorage.setItem("myLibrary", JSON.stringify(myLibrary));
+  render(myLibrary);
+}
+
+// Finds the index of book in myLibrary array.
+function getIndexofBook(bookId) {
+  let ids = myLibrary.map(current => current.id);
+  let index = ids.indexOf(bookId);
+  return index;
+}
+
+// Main function to edit the book info.
+function editBookInfo(bookId){
+  let index = getIndexofBook(bookId);
+  let classStatus = form.classList.contains('form-visibility');
+  if(classStatus == false);
+  {
+    form.classList.add('form-visibility');
+  }
+  sendInfoToForm(index);
+}
+
+function saveEditedInfo(index){
+  let changedInputs = readInputValues();
+  myLibrary[index].title = changedInputs.bookTitle;
+  myLibrary[index].author = changedInputs.authorName;
+  myLibrary[index].description = changedInputs.description;
+  myLibrary[index].pages = changedInputs.pagesNum;
+  myLibrary[index].read = changedInputs.status;
```

Comment:
we can destructure the `readInputValues()` like mentioned before, will look much cleaner